code/framework/auditing/src/main/java/io/cattle/platform/iaas/api/auditing/AuditServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package io.cattle.platform.iaas.api.auditing;
+
+import io.cattle.platform.api.auth.Identity;
+import io.cattle.platform.api.auth.Policy;
+import io.cattle.platform.core.constants.AccountConstants;
+import io.cattle.platform.iaas.api.auditing.dao.AuditLogDao;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.json.JsonMapper;
+import io.github.ibuildthecloud.gdapi.model.Resource;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuditServiceImpl implements AuditService{
+
+    @Inject
+    AuditLogDao auditLogDao;
+    @Inject
+    JsonMapper jsonMapper;
+
+    private static final Logger log = LoggerFactory.getLogger(AuditLogsRequestHandler.class);
+
+    @Override
+    public void logRequest(ApiRequest request, Policy policy) {
+        Map<String, Object> data = new HashMap<>();
+        Map requestObject = CollectionUtils.toMap(request.getRequestObject());
+        data.put("requestObject", requestObject);
+        data.put("requestParams", CollectionUtils.toMap(request.getRequestParams()));
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try {
+            jsonMapper.writeValue(os, request.getResponseObject());
+            data.put("responseObject", os.toString());
+        } catch (IOException e) {
+            log.error("Failed to write response object to log.");
+            data.put("responseObject", "{\"object\":\"Failed to convert response object to json.\"}");
+        }
+        data.put("responseCode", request.getResponseCode());
+        data.put("responseContentType", request.getResponseContentType());
+        Identity user = null;
+        for (Identity identity: policy.getIdentities()){
+            if (identity.getExternalIdType().contains("user")){
+                user = identity;
+                break;
+            }
+        }
+        if (user == null && policy.getIdentities().size() == 1){
+            user = policy.getIdentities().iterator().next();
+        }
+        long runtime = ((long) request.getAttribute("requestEndTime")) - ((long)request.getAttribute("requestStartTime"));
```

Comment:
Would requestEndTime always set on the request? I just see that in this PR you set it for only non GET methods, and wonder if there are chances for NPE when cast to (long):

if (!Schema.Method.GET.isMethod(request.getMethod())){
+            request.setAttribute("requestEndTime", System.currentTimeMillis());